### PR TITLE
History / Revisions: add loading empty states

### PR DESCRIPTION
**Pull request by danielebogo**
Refs. #10304 

This PR adds 2 empty state to the revisions list. When a user opens the revisions list for the first time, an empty state will appear while the manager is loading. If there are no revisions or if there's a problem, then the no result empty state appears. The reload button allows the user to try to load the revisions again

revisions-list-empty-states

**To test:**
- Open a revisions list for the first time. You should see the empty state.
- If an error occurs then you should see the no result empty state with the try again button

@iamthomasbishop can you let me know what do you think about the text displayed?

### CODE REVIEW

**Comment by iamthomasbishop** [comment]
My main thing on the loading states is that we should align with existing empty state patterns consistency. I'm not super familiar with _all_ of the loading/empty states that we have available, but I imagine the loading state could be pretty generic (just a spinner, or the `Loading...` label? if we don't already have another pattern to use). Imo, the books loader feels odd here. 

The empty state looks about right visually, but a couple thoughts: 

* The primary title/label of the empty state should read `No history yet` to align with Android

* We could remove the `Try again` button. I'm not sure if it adds any value or utility there.

* Worth considering: On Android, we have a message on the empty state that reads "When you make changes to your `post`/`page` you'll be able to see the history here". We could match that if having a message matches with the other empty states.

File: WordPress/Classes/ViewRelated/Post/Revisions/RevisionsTableViewController.swift [hunk 1]
```diff
@@ -76,8 +81,40 @@ private extension RevisionsTableViewController {
...
+        }
+
         manager?.getRevisions()
     }
+
+    private func configureAndDisplayNoResults(title: String,
+                                      subtitle: String? = nil,
+                                      buttonTitle: String? = nil,
+                                      accessoryView: UIView? = nil,
+                                      forNoSearchResults: Bool = false) {
```
Comment (ScoutHarris) [inline WordPress/Classes/ViewRelated/Post/Revisions/RevisionsTableViewController.swift]: It appears `subtitle` and `forNoSearchResults` are not used. Can they be removed?

**Comment by ScoutHarris** [comment]
Hey @iamthomasbishop , @danielebogo .

> My main thing these states is that we should align with existing empty state patterns consistency. 

Actually, it does. All the updated loading screens look like this, with the animated box and a message. Typically the only difference is the `Loading...` message. It's usually a bit more specific, like `Loading plans...` or `Loading plugins...`. But we do have some places that just say `Loading...`.

> Imo, the books loader feels odd here.

This can be a different discussion if you like. I just co-opted the loading screen when I updated the empty states. But it can be changed if we like.

> The primary title/label of the empty state should read `No history yet` to align with Android
> We could remove the `Try again` button. I'm not sure if it adds any value or utility there.
> Worth considering: On Android, we have a message on the empty state that reads "When you make changes to your `post`/`page` you'll be able to see the history here". We could match that if having a message matches with the other empty states.

FWIW, I agree with all these suggestions! Further, regarding the `Try again`, it's shown in two cases:
> If there are no revisions or if there's a problem

I think those should be separate. If there's no revisions, then no need to try again, and show the `When you make changes to your...` message. 

But if there's a problem, and if `Try again` will actually serve a purpose, then I can see that. 

@danielebogo I don't know what failure means in this case, but in some cases the button actually goes to support. (ex: `ActivityListViewController:actionButtonPressed`) Just something to consider.

**Comment by iamthomasbishop** [comment]
@ScoutHarris Thanks for clarifying! I wasn't super involved in the empty states so that helps me understand what currently exists.

> It's usually a bit more specific, like Loading plans... or Loading plugins.... But we do have some places that just say Loading....

I don't think a simple `Loading...` is a big problem, but it wouldn't hurt to be specific and have the label say `Loading post/page history...` in this case. If it's a pain, and it's not much gain, then let's rock as-is.

> I just co-opted the loading screen when I updated the empty states. But it can be changed if we like.

If I'm understanding you correctly (all loading screens use the "books" animation), I think it might be worth coming up with a new default, or not using the "books" as a default. It certainly works in various places like posts/pages, but not everywhere. That could be out of scope here, so if we need to move that outside this PR, let's do that.

> Further, regarding the Try again, it's shown in two cases
> If there are no revisions or if there's a problem

Thanks for this clarification. I agree w/ your assessment – show the `Try again` button only if there's an error or issue loading. If there simply are no revisions to show then no button is shown.

----

Thanks for the details!

File: WordPress/Classes/ViewRelated/Post/Revisions/RevisionsTableViewController.swift (hunk 1 above)
Comment (danielebogo) [reply WordPress/Classes/ViewRelated/Post/Revisions/RevisionsTableViewController.swift]: You are right! I removed `forNoSearchResults`. Subtitle now is used when a no result is displayed.

**Comment by danielebogo** [comment]
@ScoutHarris @nheagy @iamthomasbishop I fixed the empty states and these are the 3 cases:

revisions-list-empty-states

Let me know what you think!

**Comment by danielebogo** [comment]
@ScoutHarris I changed a bit the code to display the right empty state. Let me know what you think!

**Comment by ScoutHarris** [comment]
@danielebogo I like it! :shipit: from me.

**Comment by iamthomasbishop** [comment]
LGTM as well! :shipit:

**Comment by danielebogo** [comment]
Thanks @ScoutHarris @iamthomasbishop !

**Pull request merged**